single_level.py

```diff
@@ -90,22 +90,67 @@ def makeRow(resource):
 		else:
 			row.append("false")
 	
-	print row
+	print row	
 
 def main():
-	print "Creating a csv"
+	#User input to refine functionality of script 
+	print "Welcome to DACSspace!"
+	print ""
+	print "I'll ask you a series of questions to refine how the script works."
+	print "If you want to use the default value for a question press the ENTER key."
+	print ""
+	unpublished_response = raw_input("Do you want DACSspace to include unpublished resources? (default is no): ")
+	uniqueid_response = raw_input("Do you want to further limit the script by a specific resource id? (default is no): ")
+	
+	#Getting list of resources
+	resourceIds = requests.get(repositoryBaseURL + "/resources?all_ids=true", headers=headers)
+	
+	#Creating csv
 	writer = csv.writer(open(spreadsheet, "wb"))
-	column_headings = ["title", "resource", "extent", "date", "language", "repository", "creator", "scope", "restrictions"]
+	column_headings = ["title", "publish", "resource", "extent", "date", "language", "repository", "creator", "scope", "restrictions"]
 	writer.writerow(column_headings)
 
-	print "Getting a list of resources"
-	resourceIds = requests.get(repositoryBaseURL + "/resources?all_ids=true", headers=headers)
-	print "Writing rows"
-	for resourceId in resourceIds.json():
-			resource = (requests.get(repositoryBaseURL + "/resources/" + str(resourceId), headers=headers)).json()
-			if resource["publish"]:
+	#Checking ALL resources
+	if unpublished_response:
```

Comment:
So this just checks for the existence of a variable. Someone could respond "no" or "N" or "no way!" to the question "Do you want DACSspace to include unpublished resources?" and the way this is written now the script would export everything.
I think you need to a) tell the user what value they need to enter in order to export everything (in line 102 above) and _then_ on this line you need to check for that string. The `uniqueid_response` is different; the logic as you've written it there works fine.